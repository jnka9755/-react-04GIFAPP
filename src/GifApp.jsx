import { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifApp = () => {

  const [categories, setCategories] = useState([ 'Halo', 'Diablo' ]);

  const onAddCategory = (newCategory) =>{

    if(categories.includes(newCategory)) return;
    setCategories([ newCategory, ...categories]);
  }

  return (
    <>
        {/* Titulo */}
        <h1>Gif App</h1>

        {/* Input */}
        <AddCategory 
          // setCategories={ setCategories } 
          onNewCategory = { event => onAddCategory(event) }
        />

        {/* Listado de Gif */}
        <ol>
          { categories.map( category => { 
                    return <li key={ category }>{ category }</li>
            }) 
          }
          {/* Gif */}
        </ol>
    </>
  )
}