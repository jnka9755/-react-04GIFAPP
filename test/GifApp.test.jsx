import { render, screen, fireEvent } from "@testing-library/react";
import { GifApp } from "../src/GifApp";

describe('GifApp Test', () => {

    test('should first', () => {

        const { container } = render(
            <GifApp />
        )
        
        expect( container ).toMatchSnapshot();
    });

    test('AddCategory test', () => {

        const category = 'Halo'

        const { container } = render( <GifApp /> );
 
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
 
        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form );
 
        fireEvent.input( input, { target: { value: category + '2' } } );
        fireEvent.submit( form );
 
         expect( container.getElementsByClassName('card-grid').length ).toBe(3) ;
    });

    test('NoAddCategory test', () => {

        const category = 'Diablo'

        const { container } = render( <GifApp /> );
 
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
  
        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form ); 
 
        fireEvent.input( input, { target: { value: category } } );
        fireEvent.submit( form );
 
        expect( container.getElementsByClassName('card-grid').length ).toBe(1);
    });
});