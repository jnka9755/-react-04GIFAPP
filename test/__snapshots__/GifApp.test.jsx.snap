// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifApp Test should first 1`] = `
<div>
  <h1>
    Gif App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Diablo
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
